packages/neuron-wallet/src/services/transactions.ts

```diff
@@ -1,31 +1,65 @@
-import { transactions } from '../mock'
+import { transactions, mockedTransaction } from '../mock'
+import { Cell, Script, OutPoint } from '../cell'
 
 export interface Transaction {
   hash: string
-  type: number
-  date: number
-  value: string
+  version: number
+  deps: OutPoint[]
+  inputs?: any
+  outputs?: Cell[]
+  time?: number
+  value?: string
 }
-export interface TransactionsParams {
+
+export interface Input {
+  previousOutput: OutPoint
+  unlock: Script
+}
+export interface TransactionsByAddressesParam {
   pageNo: number
   pageSize: number
   addresses: string[]
 }
-let currentTransaction: Transaction[] = transactions
+
+export interface TransactionsByLockHashesParam {
+  pageNo: number
+  pageSize: number
+  lockHashes: string[]
+}
+
+export interface PaginationResult<T = any> {
+  totalCount: number
+  items: T[]
+}
+
+let currentTransaction: any[] = transactions
+/* eslint @typescript-eslint/no-unused-vars: "warn" */
 export default class TransactionsService {
-  public static index = (params?: TransactionsParams): Transaction[] => {
-    if (params) {
-      // TODO
+  public static getAll = (params: TransactionsByLockHashesParam): PaginationResult<Transaction> => {
+    const transaction: Transaction = mockedTransaction
+
+    return {
+      totalCount: params.pageNo * params.pageSize,
+      items: Array.from({
+        length: params.pageSize,
+      }).map(() => transaction),
```

Comment:
Transactions have the same hash will make the neuron-ui render only one record.

Reply:
Ok